### PR TITLE
Feature/goerli crosschain

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,9 +1,7514 @@
 {
   "name": "generic-oracle-dapp",
-  "version": "1.0.0",
-  "lockfileVersion": 1,
+  "version": "1.1.0",
+  "lockfileVersion": 2,
```
Comment (PierreJeanjacquot) [inline package-lock.json:4]: lockfile version upgraded (incompatible with npm < 7)

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
Before anything else, format and lint the code to track most of the common issues.
Once done see the comments.

## prettier / estlint config for TS
install deps
```
npm i --save-dev eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin prettier eslint-config-prettier eslint-plugin-prettier
```

`.eslintrc`
```
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended"
  ],
  "rules": {}
}
```

File: package.json [hunk 2]
```diff
@@ -1,35 +1,49 @@
...
   "main": "src/app.js",
   "dependencies": {
     "big.js": "^6.0.3",
-    "ethers": "^5.1.0",
     "jsonpath": "^1.1.0",
     "node-fetch": "^2.6.1",
     "path": "^0.12.7",
     "yup": "^0.32.9"
   },
   "devDependencies": {
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply package.json:15]: Is it better now?

File: tests/contractLoader.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { expect } from "chai";
```
[the remark this answers is not in the document]
Comment (PierreJeanjacquot) [reply tests/contractLoader.test.ts]: remove `chai` from dependencies

[the remark this answers is not in the document]
Comment (jeremyjams) [reply tests/contractLoader.test.ts]: Updated thanks https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/128faa07668ba7936ec56381e6b34cc00ca7b8af.

File: package.json [hunk 4]
```diff
@@ -1,35 +1,49 @@
...
+    "@jeremyjames/generic-oracle-contracts": "^1.1.0-0",
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply package.json]: Updated here https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/b3b7e2320a1e2b1c2e7c5df705fd68051f1065e9, thanks :)

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
some more minor adjustments

File: src/contractLoader.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+export class ContractLoader {
+  static loadClassicOracle(encodedArgs: string | undefined): ClassicOracle {
+    console.log("Target chain: " + chain);
+    console.log("Target oracle address: " + oracleReceiver);
+
+    if (encodedArgs == undefined) {
+      throw Error("Encoded args are required");
+    }
+    const buff = Buffer.from(encodedArgs, "base64");
+    const appDeveloperSecretsJsonString = buff.toString("utf-8");
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/contractLoader.ts:19]: OK, thanks!

File: src/dapp.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    dataType: paramSet.dataType,
+  });
+  console.log(
+    "Received response from API [url:%s, date:%s, value:%s]",
+    paramSet.url,
+    new Date(date * 1000),
+    value
+  );
+
+  const encodedValue: string | undefined = encodeValue(
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/dapp.ts]: Thanks for your NB, I see :+1: (https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/8b15022c970067030df97c8e171202d26ce3175c)

File: src/contractLoader.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    const appDeveloperSecretJson: OracleArgs = JSON.parse(
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/contractLoader.ts]: Added explicit control, thanks (https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/b18072137918445b64aa26514636f987aec6ecc2)

File: src/contractLoader.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    });
+
+    const wallet = new ethers.Wallet(targetPrivateKey, provider);
+    console.log("Target reporter wallet address: " + wallet.address); //0x767A2D69D7278F200ae1F79a00Ac2CaE299dD784
+
+    return new ClassicOracle__factory().attach(oracleReceiver).connect(wallet);
+  }
+}
+
+export class OracleArgs {
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/contractLoader.ts]: Ok, at least updated to an interface (https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/adac22a8a1eb4fff6a64140f52d0d1b5bb054e25)

File: src/dapp.ts [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+      encodedValue
+    );
+    console.log(
+      "Sent transaction to targeted oracle [tx:%s, oracleId:%s, encodedValue:%s]",
+      tx.hash,
+      oracleId,
+      encodedValue
+    );
+    const receipt = await tx.wait();
+    if (receipt.blockNumber) {
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/dapp.ts]: Thank you, updated here: https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/13/commits/07ef14a4e59530787a1a5f48447fd5af54ba56e2

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**